### PR TITLE
BlobDB interface changes/cleanup

### PR DESCRIPTION
**Pull request by dannyroberts**
@millerdev is this pretty much what you were thinking when we talked this morning? @snopoke fyi. I think maybe most important thing is if you have a sense of whether I found all uses of `AbstractBlobDB.put`, because if there are any that PyCharm's Find Usages missed that pass in basename, then they'll fail. (And `put` is kind of a hard thing to just grep for, unfortunately!)

### CODE REVIEW

**Comment by dannyroberts** [comment]
@czue buddy

**Review by czue — APPROVED** [review]
code looks good but i'd feel more comfortable if @millerdev or @snopoke also did a pass since I don't have much context

File: corehq/blobs/atomic.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def __init__(self, db):
     def put(self, content, basename="", bucket=DEFAULT_BUCKET):
```
Comment (millerdev) [inline corehq/blobs/atomic.py]: I think `basename` should be removed here as well.

File: corehq/blobs/interface.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ class AbstractBlobDB(object):
     __metaclass__ = ABCMeta
 
     @abstractmethod
-    def put(self, content, basename="", bucket=DEFAULT_BUCKET):
+    def put(self, content, bucket=DEFAULT_BUCKET, identifier=None):
```
Comment (millerdev) [inline corehq/blobs/interface.py]: The docstring for this method needs to be updated.

File: corehq/blobs/atomic.py (hunk 1 above)
Comment (dannyroberts) [inline corehq/blobs/atomic.py]: Good call

File: corehq/blobs/interface.py (hunk 2 above)
Comment (dannyroberts) [inline corehq/blobs/interface.py]: Good catch, will fix

**Review by millerdev — APPROVED** [review]

**Comment by dannyroberts** [comment]
@millerdev what's your feeling here; is there anything we need to test manually that wouldn't be caught by the travis build?

**Comment by millerdev** [comment]
Don't think so. I think g2g when you're happy with it. FWIW, I checked out this branch locally and did a quick search for `.put(` and reviewed all occurrences, most of which are blob db puts. Nothing jumped out at me as needing attention other than the things you already addressed.

**Comment by dannyroberts** [comment]
Awesome, I'm happy with it then so I'll go ahead and merge it!

**Pull request merged**